### DISCUSSION TITLE
vaultsharp support for vault on K8s

### DISCUSSION THREAD

**Discussion by maharajanm** [rajanadar/VaultSharp#202] [Q&A] [answered]
Hi, 
Please clarify my use case below. 

Have a vault running as an HA cluster on a dedicated K8s cluster. 
I have both .Net monolith(running on IIS) and .Net microservices(running as a pod on K8s cluster) and I'm looking at `vaultsharp ` lib to integrate in both of them, so that both the type of apps should be able to access the secrets stored in the vault running on K8s. Does `vaultsharp `support vault running on K8s. 

Thanks in advance.

**Comment by rajanadar** [comment, marked as answer]
Hi @mahadreamz

VaultSharp has nothing to do with k8s or docker swarm or containerization tech. All it cares about is the dot net app using it and the dot net version. And VaultSharp supports a wide variety of for net versions. 

And finally in terms of the Vault Server itself, VaultSharp doesn't care where and how the Vault api is hosted. Ecs, eks, pods whatever. As long as the vault api is accessible from your app, VaultSharp will do its thing.

Let me know if anymore questions.

**Reply by maharajanm** [reply]
Hi @rajanadar 
Thanks for the quick response. 

If I understand correctly, `vaultsharp `doesn't care where the vault is hosted either on K8s cluster or just server, as long as the vault is exposed as a service at a given IP and the .Net app (micro or monolith) with `vaultsharp `integrated should be able to access the vault.

**Reply by rajanadar** [reply]
Perfect.

**Reply by maharajanm** [reply]
Hi @rajanadar 

Another clarification around the integration of `vaultsharp`(vs) . Not sure, though whether I can post this here or not. 

I followed the tutorial to integrate the vs with my .net service app and accessed the vault secrets successfully. 

But given the 'n' number of microservice use cases, should I have to add all the vs client code (integration code to access and inject) that must be written in all the applications. Basically, here, I have to duplicate the same in every microservice app. Is there any better approach to solve this problem.

**Reply by rajanadar** [reply]
Probably not. Plus I wouldn't call it duplication. It is the bare minimum thing. 

Every micro service has to deal with logging, secret secret management, connection pool settings, redis cache initialization, database context initialization, oauth filters etc. We don't necessarily look at them as duplicated functionality.

Also, to get real, a VaultSharp integration is probably 3-4 lines of code to inject a IVaultClient. So any other option maybe even more work.

**Reply by maharajanm** [reply]
That makes a lot of sense. Thank you and I appreciate your prompt response.

**Reply by rajanadar** [reply]
Cheers
